src/ray/raylet/scheduling/scheduling_ids.cc

```diff
@@ -37,38 +39,56 @@ std::string StringIdMap::Get(uint64_t id) const {
 };
 
 int64_t StringIdMap::Insert(const std::string &string_id, uint8_t max_id) {
-  auto sit = string_to_int_.find(string_id);
-  if (sit == string_to_int_.end()) {
-    int64_t id = hasher_(string_id);
-    if (max_id != 0) {
-      id = id % MAX_ID_TEST;
+  auto id = Get(string_id);
```

Comment:
We can add coarse grained write lock and optimize later (if necessary)

Reply:
> We can add a single coarse grained write lock and optimize later (if necessary)

Ok, it will be more simple.